scrapper.py

```diff
@@ -31,67 +42,260 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+    def __init__(self, seed_urls: list, total_max_articles: int, max_articles_per_seed: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.max_articles_per_seed = max_articles_per_seed
+        self.urls = []
 
     @staticmethod
-    def _extract_url(article_bs):
-        pass
+    def _extract_url(article_bs, seen):
+        extracted = list({link['href'] for link in article_bs.find_all('a', href=True)})
+        return list(filter(lambda x: x.startswith('/news/')
+                           and x not in seen, extracted))
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for link in self.seed_urls:
+            article_bs = BeautifulSoup(requests.get(link, 'html.parser').text, 'html.parser')
+            newfound = self._extract_url(article_bs, self.urls)
+            self.urls.extend(newfound[:self.max_articles_per_seed])
+        self.urls = [i for i in self.urls if len(i) > 20
+                     and not any(map(lambda y: y.isupper(), i))][:self.total_max_articles]
+        print('Scraped seed urls, overall number of urls is', len(self.urls))
+        old = len(self.urls)
+        while len(self.urls) < self.total_max_articles:
+            print('Due to insufficient number started further iteration')
+            print('current number', len(self.urls), ', required', self.total_max_articles)
+            for link in self.urls:
+                article_bs = BeautifulSoup(requests.get(URL_START + link, 'html.parser').text, 'html.parser')
+                newfound = list(filter(lambda x: len(x) > 20, self._extract_url(article_bs, self.urls)))
+                print('    checked new url, found', len(newfound), 'articles')
+                self.urls.extend(newfound[:self.max_articles_per_seed])
```

Comment:
because I thought of the possibility of the task that imposes restrictions concerning how many urls one should get from single page so that the sample of the scraped articles is more representative (as, we all know, recommendations on the page are likelier to contain links to the works of the same author or on the same theme)

so, for example, if i wish to avoind getting imbalanced dataset in which 80% of articles will belong to the same author, I might want to scrape less links from each page

is it reasonable?